Exception should be raised if searching with invalid params

**Please state your issue using the following template and, most importantly, in English.**

**Describe the bug**
```
======================================================================================== FAILURES =========================================================================================
_________________________________________________ TestSearchInvalid.test_search_with_invalid_params[get_simple_index0-get_search_params0] _________________________________________________

self = <test_search.TestSearchInvalid object at 0x7f6076724da0>, connect = <milvus.client.stub.Milvus object at 0x7f607672e240>, collection = 'search_siKb91qs'
get_simple_index = {'index_type': 'FLAT', 'metric_type': 'L2', 'params': {'nlist': 128}}, get_search_params = {'index_type': 'IVF_FLAT', 'search_params': {'invalid_key': 100}}

    @pytest.mark.level(2)
    def test_search_with_invalid_params(self, connect, collection, get_simple_index, get_search_params):
        '''
        target: test search fuction, with the wrong nprobe
        method: search with nprobe
        expected: raise an error, and the connection is normal
        '''
        search_params = get_search_params
        index_type = get_simple_index["index_type"]
        entities, ids = init_data(connect, collection)
        connect.create_index(collection, field_name, get_simple_index)
        query, vecs = gen_query_vectors(field_name, entities, top_k, 1, search_params=search_params["search_params"])
        with pytest.raises(Exception) as e:
>           res = connect.search(collection, query)
E           Failed: DID NOT RAISE <class 'Exception'>


```

**Steps/Code to reproduce behavior**
Follow this [guide](http://matthewrocklin.com/blog/work/2018/02/28/minimal-bug-reports) to craft a minimal bug report. This helps us reproduce the issue you're having and resolve the issue more quickly.

**Expected behavior**
A clear and concise description of what you expected to happen.

**Environment details**
- Hardware/Softward conditions (OS, CPU, GPU, Memory)
- Method of installation (Docker, or from source)
- Milvus version (v0.3.1, or v0.4.0)
- Milvus configuration (Settings you made in `server_config.yaml`)

**Screenshots**
If applicable, add screenshots to help explain your problem.

**Additional context**
Add any other context about the problem here.

--- a/core/src/index/knowhere/knowhere/index/vector_index/ConfAdapter.cpp
+++ b/core/src/index/knowhere/knowhere/index/vector_index/ConfAdapter.cpp
@@ -110,13 +110,13 @@ IVFConfAdapter::CheckTrain(Config& oricfg, const IndexMode mode) {
 
 bool
 IVFConfAdapter::CheckSearch(Config& oricfg, const IndexType type, const IndexMode mode) {
-    if (mode == IndexMode::MODE_GPU) {
+    int64_t max_nprobe = MAX_NPROBE;
 #ifdef MILVUS_GPU_VERSION
-        CheckIntByRange(knowhere::IndexParams::nprobe, MIN_NPROBE, faiss::gpu::getMaxKSelection());
-    } else {
-#endif
-        CheckIntByRange(knowhere::IndexParams::nprobe, MIN_NPROBE, MAX_NPROBE);
+    if (mode == IndexMode::MODE_GPU) {
+        max_nprobe = faiss::gpu::getMaxKSelection();
     }
+#endif
+    CheckIntByRange(knowhere::IndexParams::nprobe, MIN_NPROBE, max_nprobe);
 
     return ConfAdapter::CheckSearch(oricfg, type, mode);
 }

